### ISSUE TITLE
Generated parsers should use a custom error type for Result

### ISSUE THREAD

**Issue by johnpmayer** [kevinmehall/rust-peg#57]
When using the try! macro around a generated parser from within a function which returns another Result with a different custom error... 

```Rust
fn foo() -> Result<(), MyCustomError> {
  let value = try!(grammar::expr("xxx"));
  ...
  Ok(())
}
```

...because the generated parsers use String for Result errors, I must implement the following for my custom error type.

```Rust
impl FromError<String> for MyCustomError {
  ...
}
```

In the current situation, I have two choices

#### Option 1

```Rust
pub enum MyCustomError { IoError(IoError), ParseError(String) }
```

Send all String to ParseError. This is bad if there are other libraries which also use String for an error; the semantics of FromError<String> do not make sense. (I'd raise the same issues to the other library for using String). It's an error-space collision!

#### Option 2 

```Rust
pub enum MyCustomError { IoError(IoError), GenericError(String) }
```

Send all String to GenericError, which loses the ability to have specificity, such as custom formatting in the implementation of FromError<String>.

#### Proposal

Have parsers generated by rust-peg instead look something like

```Rust
fn expr(&str) -> Result<a, ParseError>
```

**Comment by kevinmehall** [comment]
Agreed, it should be a ParseError enum containing machine-readable error information, with a String or Debug implementation to format it to a human-readable error if desired.

**Comment by kevinmehall** [comment]
This was fixed by #60 / 0.2.0 thanks to @jonas-schievink.
